python/cudf/cudf/core/series.py

```diff
@@ -171,10 +174,25 @@ def _append_new_row_inplace(col: ColumnBase, value: ScalarLike) -> None:
     """
     val_col = as_column(value, dtype=col.dtype if value is None else None)
     to_type = find_common_type([val_col.dtype, col.dtype])
+    if (
+        cudf.get_option("mode.pandas_compatible")
+        and is_string_dtype(to_type)
+        and is_mixed_with_object_dtype(val_col, col)
+    ):
+        raise MixedTypeError("Cannot append mixed types")
     val_col = val_col.astype(to_type)
     old_col = col.astype(to_type)
-
-    col._mimic_inplace(concat_columns([old_col, val_col]), inplace=True)
+    res_col = concat_columns([old_col, val_col])
+    if (
+        cudf.get_option("mode.pandas_compatible")
+        and res_col.dtype != col.dtype
+        and isinstance(col, CategoricalColumn)
```

Comment:
```suggestion
        and isinstance(col.dtype, CategoricalDtype)
```

Reply:
Also, I imagine we would want this in `concat_column`? We wouldn't be able to represent this type properly with any type of column concatenation?

Reply:
I'm trying to make this change future-proof so that we can support `pd.ArrowDtype(pa.dict(..))`.

Reply:
> Also, I imagine we would want this in `concat_column`? We wouldn't be able to represent this type properly with any type of column concatenation?

I have another series of fixes coming to concat, I think this will get moved to there. I'm trying to keep this PR changes as minimal and targeted as possible.